### PR TITLE
Various mappings for net/minecraft/world/level/levelgen

### PR DESCRIPTION
**Pull request by PiTheGuy**

### CODE REVIEW

**Comment by PiTheGuy** [comment]
Converted to draft while I update this to 1.19

**Comment by PiTheGuy** [comment]
I changed my mind. I have no idea what I'm doing.

File: data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping [hunk 1]
```diff
@@ -3,12 +3,20 @@ CLASS net/minecraft/world/level/levelgen/SurfaceSystem
 		ARG 1 noises
 		ARG 2 defaultBlock
 		ARG 3 seaLevel
+	METHOD buildSurface (Lnet/minecraft/world/level/biome/BiomeManager;Lnet/minecraft/core/Registry;ZLnet/minecraft/world/level/levelgen/WorldGenerationContext;Lnet/minecraft/world/level/chunk/ChunkAccess;Lnet/minecraft/world/level/levelgen/NoiseChunk;Lnet/minecraft/world/level/levelgen/SurfaceRules$RuleSource;)V
+		ARG 4 context
+		ARG 5 chunk
+		ARG 6 noiseChunk
+	METHOD generateBands (Lnet/minecraft/world/level/levelgen/RandomSource;)[Lnet/minecraft/world/level/block/state/BlockState;
+		ARG 0 randomSource
```
Comment (sciwhiz12) [inline data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping]: Current convention is to name `RandomSource` parameters as `random`.
```suggestion
		ARG 0 random
```

File: data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping [hunk 2]
```diff
@@ -3,12 +3,20 @@ CLASS net/minecraft/world/level/levelgen/SurfaceSystem
...
 	METHOD getOrCreateNoise (Lnet/minecraft/resources/ResourceKey;)Lnet/minecraft/world/level/levelgen/synth/NormalNoise;
 		ARG 1 paramaters
 	METHOD getOrCreateRandomFactory (Lnet/minecraft/resources/ResourceLocation;)Lnet/minecraft/world/level/levelgen/PositionalRandomFactory;
 		ARG 1 location
 	METHOD isStone (Lnet/minecraft/world/level/block/state/BlockState;)Z
 		ARG 1 state
+	METHOD makeBands (Lnet/minecraft/world/level/levelgen/RandomSource;[Lnet/minecraft/world/level/block/state/BlockState;ILnet/minecraft/world/level/block/state/BlockState;)V
+		ARG 0 randomSource
```
Comment (sciwhiz12) [inline data/net/minecraft/world/level/levelgen/SurfaceSystem.mapping]:
```suggestion
		ARG 0 random
```

File: data/net/minecraft/world/level/levelgen/structure/placement/RandomSpreadType.mapping [hunk 3]
```diff
@@ -3,3 +3,5 @@ CLASS net/minecraft/world/level/levelgen/structure/placement/RandomSpreadType
 		ARG 3 id
 	METHOD byName (Ljava/lang/String;)Lnet/minecraft/world/level/levelgen/structure/placement/RandomSpreadType;
 		ARG 0 name
+	METHOD evaluate (Lnet/minecraft/world/level/levelgen/RandomSource;I)I
+		ARG 1 randomSource
```
Comment (sciwhiz12) [inline data/net/minecraft/world/level/levelgen/structure/placement/RandomSpreadType.mapping]:
```suggestion
		ARG 1 random
```

**Review by sciwhiz12 — CHANGES_REQUESTED** [review]

**Review by sciwhiz12 — APPROVED** [review]

**Pull request merged**